tests/test_tar.py

```diff
@@ -842,31 +825,6 @@ def test_outer_tar_must_not_be_compressed(tmp_path: Path) -> None:
                 pass
 
 
-@pytest.mark.parametrize(
-    "format", [tarfile.PAX_FORMAT, tarfile.GNU_FORMAT, tarfile.USTAR_FORMAT]
-)
-def test_tar_stream(tmp_path: Path, format: int) -> None:
-    # Create Tarfile
-    main_tar = tmp_path.joinpath("backup.tar")
-
-    with patch.object(tarfile, "DEFAULT_FORMAT", format):
-        ostf = SecureTarFile(main_tar, "w", gzip=False)
-        with ostf as tar_file:
-            tar_info = tarfile.TarInfo(name="test.txt")
-            with _add_stream(tar_file, tar_info, ostf) as stream:
```

Comment:
`_add_stream` is moved to `_InnerSecureTarFile` and the corresponding functionality can't be called directly

I think removing this test is fine since there are multiple tests calling `TarFile.create_inner_tar`

Reply:
On second thought, it seems the test intended to make sure we handle all the tar header variants correctly.
I think it didn't do that correctly though, since we do some magic here, which already before this PR was not in the `_add_stream` function: https://github.com/home-assistant-libs/securetar/blob/23ec55c3b1a5e212f8d9fd01c7b4222c4b821855/securetar/__init__.py#L608-L619

I'll restore the removed test.